spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/support/JdbcQueryMethod.java

```diff
@@ -53,13 +59,49 @@ public JdbcQueryMethod(Method method, RepositoryMetadata metadata, ProjectionFac
 	 *
 	 * @return May be {@code null}.
 	 */
+
 	@Nullable
-	String getAnnotatedQuery() {
+	public String getAnnotatedQuery() {
+		String annotatedValue = getQueryValue();
+		return StringUtils.hasText(annotatedValue) ? annotatedValue : getNamedQuery();
+	}
+
+	/**
+	 * Returns the annotated query with key value if it exists.
+	 *
+	 * @return May be {@code null}.
+	 */
+	@Nullable
+	public String getQueryValue() {
 		return getMergedAnnotationAttribute("value");
 	}
 
+
+	/**
+	 * Returns the annotated query name.
+	 *
+	 * @return May be {@code null}.
+	 */
+	@Nullable
+	public String getQueryName() {
+		return getMergedAnnotationAttribute("name");
+	}
 	/**
-	 * Returns the class to be used as {@link org.springframework.jdbc.core.RowMapper}
+	 * Returns the annotated query with key name if it exists.
+	 *
+	 * @return May be {@code null}.
+	 */
+	@Nullable
+	public String getNamedQuery() {
+		String annotatedName = getMergedAnnotationAttribute("name");
+		return (StringUtils.hasText(annotatedName) && this.namedQueries.hasQuery(annotatedName))
+				? this.namedQueries.getQuery(annotatedName)
+				: super.getName();
```

Comment:
We can't return the name of the method as a query, we have to use it as the query name to look up a query.

Reply:
where is looking up the query?,  in the properties file?.

Reply:
Exactly.